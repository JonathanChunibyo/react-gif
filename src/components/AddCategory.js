import React, { useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({ setcategory }) => {
    const vacio = () => {
        setinputValue('');
    }
    const [inputValue, setinputValue] = useState('');
    const handleInputvalue = (e) => {
        setinputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setcategory(cats => [inputValue, ...cats]);
            setinputValue('');
        }
    }
    return (

        <form onSubmit={handleSubmit}>

            <h1> {inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputvalue}
                onClick={vacio}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setcategory: PropTypes.func.isRequired ,
}

