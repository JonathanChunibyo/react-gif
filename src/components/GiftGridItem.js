import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__fadeInLeft'>
            <p>{title}</p>
            <img src= {url} alt={title}/>
        </div>
    )
}
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired ,
    url: PropTypes.string.isRequired,
}
