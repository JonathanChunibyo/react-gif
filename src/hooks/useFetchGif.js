
import React, { useEffect, useState } from 'react'
import { getGif } from '../helpers/getGifs';

export const useFetchGif = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true,
    })

    useEffect(() => {

        getGif(category)
        .then( img => {
           setTimeout(() => {
            setstate({
                data: img,
                loading: false
            })
           }, 3000);
        })
    }, [category]);

    return state;
}
