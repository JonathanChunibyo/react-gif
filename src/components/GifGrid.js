import React /* { useEffect, useState } */ from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
//import { getGif } from '../helpers/getGifs';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from "prop-types";


export const GifGrid = ({ category }) => {


    const {data: images, loading} = useFetchGif(category);


    
    
    /* const [images, setimages] = useState([]);
    useEffect(() => {
        getGif(category).then( setimages );
    }, [category]);
 */

    return (
        <>

            <h3 className='animate__animated animate__zoomInDown'> {category} </h3>
            {loading && <p className='animate__animated animate__shakeX'>'Cargando...'</p>}
            
            <div className='cardgrid'>
                {
                     //images.map(img => (
                       // <li key={img.id}>{img.title}</li>
                    //)) 
                    images.map((img) =>
                        <GiftGridItem
                            key={img.id}
                            {...img}
                        />)
                }
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}