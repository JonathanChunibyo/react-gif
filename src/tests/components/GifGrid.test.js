import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import {useFetchGif} from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif');
/* {
    data: [],
    loading: true,
} */

const category = 'One Punch';
const gif = [{
    id: 'BCA',
    url: 'www.book.com',
    title: 'cosa',
},{
    id: 'ABC',
    url: 'www.facebook.com',
    title: 'cualquier cosa',
}];


describe('evaluar funcionalidad de <GifGrid />', () => {


    

    test(' toMatchSnapshot - muestre lo que esta en la page ', () => {

        useFetchGif.mockReturnValue({
            data: gif,
            loading: false,
        });
    
        const wraper = shallow(<GifGrid category={category}/>);

        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').exists()).toBe(false);
        expect(wraper.find('GiftGridItem').length).toBe(gif.length);
    })

    test('debe mostrar items cuando se carguen imagenes useFetch ', () => {

    })


})
