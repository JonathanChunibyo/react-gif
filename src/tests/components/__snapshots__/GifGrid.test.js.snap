// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`evaluar funcionalidad de <GifGrid />  toMatchSnapshot - muestre lo que esta en la page  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomInDown"
  >
     
    One Punch
     
  </h3>
  <div
    className="cardgrid"
  >
    <GiftGridItem
      id="BCA"
      key="BCA"
      title="cosa"
      url="www.book.com"
    />
    <GiftGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="www.facebook.com"
    />
  </div>
</Fragment>
`;
